lib/podlet.js

```diff
@@ -288,49 +290,61 @@ const PodiumPodlet = class PodiumPodlet {
     }
 
     async process(incoming) {
-        incoming.view = this._view;
-        incoming.name = this.name;
-        incoming.css = this.cssRoute;
-        incoming.js = this.jsRoute;
+        const s = incoming;
+        s.view = this._view;
+        s.name = this.name;
+        s.css = this.cssRoute;
+        s.js = this.jsRoute;
 
         // Determine if request comes from layout server or not
         if (
-            incoming.request.headers['user-agent'] &&
-            incoming.request.headers['user-agent'].startsWith('@podium/client')
+            s.request.headers['user-agent'] &&
+            s.request.headers['user-agent'].startsWith('@podium/client')
         ) {
-            incoming.development = false;
+            s.development = false;
         } else {
-            incoming.development = this.development;
+            s.development = this.development;
         }
 
         // Append development context and proxy
-        if (incoming.development) {
-            incoming.context = Object.assign(
+        if (s.development) {
+            s.context = Object.assign(
                 this.baseContext,
-                { mountOrigin: incoming.url.origin },
-                this.defaultContext
+                { mountOrigin: s.url.origin },
+                this.defaultContext,
             );
             this.log.debug(
                 `Appending a default context to inbound request "${JSON.stringify(
-                    incoming.context
-                )}"`
+                    s.context,
+                )}"`,
             );
         } else {
-            incoming.context = utils.deserializeContext(incoming.request.headers);;
+            s.context = utils.deserializeContext(incoming.request.headers);
             this.log.debug(
-                `Inbound request contains a context "${JSON.stringify(incoming.context)}"`
+                `Inbound request contains a context "${JSON.stringify(
+                    incoming.context,
+                )}"`,
             );
         }
 
         return incoming;
     }
 
+    render(incoming, data) {
+        if (!this.development) {
+            if (typeof data === 'string') return data;
+            return data.body || '';
+        }
+
+        return incoming.render(data);
+    }
+
     middleware() {
         return (req, res, next) => {
             const incoming = new HttpIncoming(req, res);
 
             this.process(incoming)
-                .then(async (incom) => {
+                .then(async incom => {
```

Comment:
We can turn this promise chain into plain `async`/`await`.

Reply:
done